From a63d86b3f4eb362d43e970dbda528122385dc5f3 Mon Sep 17 00:00:00 2001
From: Yuchou Gan <yuchou.gan@nxp.com>
Date: Fri, 11 May 2018 23:53:27 +0800
Subject: [PATCH 3821/5242] MLK-18283-3 gpu: imx: dpu-blit: sync for cmd
 sequence finished

commit  d41e281b216801bf66c6f459760507bb2acb6f88 from
https://source.codeaurora.org/external/imx/linux-imx.git

As blit engine use cmd sequence and disabled hardware interrupt,
sync to wait the cmd sequence idle before execute new cmd sequence

Signed-off-by: Yuchou Gan <yuchou.gan@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/imx/dpu-blit/dpu-blit.c |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/gpu/imx/dpu-blit/dpu-blit.c b/drivers/gpu/imx/dpu-blit/dpu-blit.c
index bac49db..ef7162a 100644
--- a/drivers/gpu/imx/dpu-blit/dpu-blit.c
+++ b/drivers/gpu/imx/dpu-blit/dpu-blit.c
@@ -121,6 +121,8 @@ void dpu_be_configure_prefetch(struct dpu_bliteng *dpu_be,
 	}
 
 	dpu_be_wait(dpu_be);
+	dpu_cs_wait_idle(dpu_be);
+	udelay(10);
 
 	if (need_handle_start) {
 		dprc_first_frame_handle(dprc);
-- 
1.7.9.5

